packages/react/src/components/ContainedList/ContainedList.tsx

```diff
@@ -5,14 +5,51 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { ReactNode } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { LayoutConstraint } from '../Layout';
 import { useId } from '../../internal/useId';
 import { usePrefix } from '../../internal/usePrefix';
 
-const variants = ['on-page', 'disclosed'];
+const variants: ('on-page' | 'disclosed')[] = ['on-page', 'disclosed'];
```

Comment:
Is there a way to use this root definition for both the proptypes and the TS interface? Something like this maybe

```js
const variants = ['on-page', 'disclosed'] as const;

// I think this will be typed as 'on-page' | 'disclosed' because the typeof operator returns the inferred type?
type Variants = typeof variants[number]; 


// ...then below in the interface:

kind?: Variants;
```